src/lib.rs

```diff
@@ -266,29 +290,47 @@ impl Config {
         self
     }
 
-    fn is_flag_supported(&mut self, flag: &str) -> io::Result<bool> {
+    fn is_flag_supported(&mut self, flag: &str) -> Result<bool, Error> {
```

Comment:
This function calls `get_target()` and `try_get_compiler()`, in which at least `get_target()` tries to determine if an environment variable exists. The `std::env::var()` function it calls doesn't return any kind of `io::Error` (it used to panic if there the result was `None`), so I don't think its possible to use this?